lib/topology/platforms/shell.py

```diff
@@ -510,27 +517,26 @@ def __init__(
             **kwargs
         )
 
-    def _setup_connection(self, connection=None):
+    def _setup_shell(self, connection=None):
```

Comment:
Better to add some explanation in this function documentation on why this name is kept just to avoid backward-compatibility troubles.